### ISSUE TITLE
How to show feature names in plot_importance?

### ISSUE THREAD

**Issue by jcoding2022** [microsoft/LightGBM#6146]
Plot_importance gives us a plot like [this](https://i0.wp.com/neptune.ai/wp-content/uploads/2022/10/lgb-plot_importance.png?resize=768%2C572&ssl=1). Is there a way to show feature names instead of Column_X?

**Comment by jcoding2022** [comment]
> Hey @jcoding2022, thanks for using LightGBM. The labels are taken from the feature names of the trained model. For numpy arrays you can provide them through the `feature_name` argument of the [Dataset](https://lightgbm.readthedocs.io/en/stable/pythonapi/lightgbm.Dataset.html#lightgbm.Dataset) if you're using the training API or through the `feature_name` argument of the [fit method](https://lightgbm.readthedocs.io/en/stable/pythonapi/lightgbm.LGBMRegressor.html#lightgbm.LGBMRegressor.fit) if you're using the scikit-learn API. If you're using pandas they're taken from the column names automatically.
> 
> If you're just looking to update that plot you can do something like this:
> 
> ```python
> ax = lgb.plot_importance(model)
> ax.set_yticklabels(feature_names)
> ```
> 
> Please let us know if you have further doubts.

Thanks for your help! Could you provide an example for the Dataset approach and the fit method approach? I tried the fit method, and got the following error:
fit() got an unexpected keyword argument: feature_name

For the last method you showed, it worked fine if I plot all the features. What if I only want to plot the top 10 most important features? I used the following code and got an error:
```
ax = lgb.plot_importance(model, max_num_features=10)
ax.set_yticklabels(feature_list)
```
Do I have to specify the names of these 10 features?

**Comment by jmoralez** [comment]
Sure, here's an example:
```python
import lightgbm as lgb
from sklearn.datasets import make_regression

X, y = make_regression()
feature_name = [f'x_{i}' for i in range(X.shape[1])]
params = {'num_leaves': 5, 'num_iterations': 2}

# training API
ds = lgb.Dataset(X, y, feature_name=feature_name)
bst = lgb.train(params, ds)
lgb.plot_importance(bst)

# scikit-learn API
model = lgb.LGBMRegressor(**params).fit(X, y, feature_name=feature_name)
lgb.plot_importance(model)
```
I recommend using either of these approaches instead of modifying the plot, because if you want to limit the number of features you have to sort them by their importance and provide only those names. You could also manually create that plot by getting the feature importances from the model, either `bst.feature_importance()` or `model.feature_importances_` and combining them with your feature names.

**Comment by jcoding2022** [comment]
Thank you so much! The two approaches worked for me!

**Comment by jmoralez** [comment]
Glad to hear. Closing since the issue was solved.
